### PR TITLE
Added AspNetCore extension following the Owin version

### PR DESCRIPTION
**Pull request by edwardwilson**
#379 Adds new project for Asp.Net Core Metadata

### CODE REVIEW

**Comment by edwardwilson** [comment]
@rasmus seems to be an issue with sql tests in appveyor, have you seen this before?

**Comment by rasmus** [comment]
@edwardwilson yes, I even created a ticket to appveyor. But as I have been unable to reproduce it, I closed it again. It seems like the SQL Express instance isn't started in some cases, but AppVeyor said they hadn't heard about that before, so it might be something with the EventFlow MSSQL tests.

**Comment by edwardwilson** [comment]
@rasmus All tests are passing now, I think this can be reviewed and merged!

**Comment by mattgi** [comment]
@edwardwilson is this PR required to configure EventFlow services inside an ASP.NET Core 2.0 Startup file (register services, etc.)?

I can't find any reference on how to achieve this other than this PR referencing Core so thought I'd reach out to see if you can provide some guidance in getting started with a simple Web API setup.

File: Source/EventFlow.AspNetCore/EventFlow.AspNetCore.csproj [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <RepositoryUrl>https://github.com/eventflow/EventFlow</RepositoryUrl>
+    <PackageProjectUrl>http://docs.geteventflow.net/</PackageProjectUrl>
+    <PackageIconUrl>https://raw.githubusercontent.com/eventflow/EventFlow/develop/icon-128.png</PackageIconUrl>
+    <PackageLicenseUrl>https://raw.githubusercontent.com/eventflow/EventFlow/develop/LICENSE</PackageLicenseUrl>
+    <NeutralLanguage>en-US</NeutralLanguage>
+    <PackageReleaseNotes>UPDATED BY BUILD</PackageReleaseNotes>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore.Http" Version="1.1.0" />
```
Comment (mattgi) [inline Source/EventFlow.AspNetCore/EventFlow.AspNetCore.csproj]: I tried to pull this down locally (I am running VS for Mac and Core 2.0) and ran in to issues. Ended up removing `net451`, and replacing `netstandard1.6` with `netcoreapp2.0` and using the 2.0 version of `AspNetCore.Http`. I don't know enough about cross platform builds, but I would be surprised if netstandard supported AspNetCore.*. Maybe it does though and I'm just doing something wrong.

**Comment by rasmus** [comment]
@edwardwilson Are you blocked by this? I would really like if we could move it into a separate repo

**Comment by edwardwilson** [comment]
@rasmus I need to do some more testing with .net core 2.0, everything is working fine for me in 1.1 and 1.0. We could create this as it's own repo and open a bug for 2.0?

**Comment by edwardwilson** [comment]
@rasmus this is now ready to be merged; work is complete

File: Source/EventFlow.AspNetCore/MetadataProviders/AddRequestHeadersMetadataProvider.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+namespace EventFlow.Aspnetcore.MetadataProviders
+{ 
+    public class AddRequestHeadersMetadataProvider : IMetadataProvider
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        private static readonly ISet<string> RequestHeadersToSkip = new HashSet<string>
+            {
+                "Authorization",
+                "Cookie"
```
Comment (rasmus) [inline Source/EventFlow.AspNetCore/MetadataProviders/AddRequestHeadersMetadataProvider.cs]: Needs to be case-insensitive, HTTP 2 sends lower case headers

File: Source/EventFlow.AspNetCore/MetadataProviders/AddUserHostAddressMetadataProvider.cs [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            where TAggregate : IAggregateRoot<TIdentity>
+            where TIdentity : IIdentity
+        {
+            yield return new KeyValuePair<string, string>("remote_ip_address", _httpContextAccessor.HttpContext.Connection.RemoteIpAddress?.ToString());
+
+            var headerInfo = HeaderPriority
+                .Select(h =>
+                    {
+                        StringValues value;
+                        var address = _httpContextAccessor.HttpContext.Request.Headers.TryGetValue(h, out value)
```
Comment (rasmus) [inline Source/EventFlow.AspNetCore/MetadataProviders/AddUserHostAddressMetadataProvider.cs]: Is `HttpContext.Request.Headers.TryGetValue` case insensitive? HTTP 2 sends lower case headers

Comment (edwardwilson) [reply Source/EventFlow.AspNetCore/MetadataProviders/AddUserHostAddressMetadataProvider.cs]: Changes made so all checks should be case insensative

**Pull request merged**